packages/core/src/2d/assembler/SimpleSpriteAssembler.ts

```diff
@@ -74,15 +82,25 @@ export class SimpleSpriteAssembler {
     vertices[offset + 28] = top;
   }
 
-  static updateColor(renderer: SpriteRenderer): void {
+  static updateColor(renderer: SpriteRenderer, alpha: number = 1): void {
     const subChunk = renderer._subChunk;
     const { r, g, b, a } = renderer.color;
+    const finalAlpha = a * alpha;
     const vertices = subChunk.chunk.vertices;
     for (let i = 0, o = subChunk.vertexArea.start + 5; i < 4; ++i, o += 9) {
       vertices[o] = r;
       vertices[o + 1] = g;
       vertices[o + 2] = b;
-      vertices[o + 3] = a;
+      vertices[o + 3] = finalAlpha;
+    }
+  }
+
+  static updateAlpha(renderer: SpriteRenderer, alpha: number = 1): void {
+    const subChunk = renderer._subChunk;
+    const finalAlpha = renderer.color.a * alpha;
+    const vertices = subChunk.chunk.vertices;
```

Comment:
Above three lines can be optimized using destructuring to make it more concise and avoid repeated references to renderer and subChunk. Here's a suggested improvement:
const { _subChunk: { chunk: { vertices } }, color: { a } } = renderer;
const finalAlpha = a * alpha;

Reply:
I personally think this is not good for readability.